### PR TITLE
Initial docs of audio.jl and some more tests

### PR DESCRIPTION
**Pull request by ashwanirathee**
- [x] Docs in audio.jl
- [x] Tests for pitchshift, speedup, slowdown, zero crossing rate

### CODE REVIEW

**Comment by ashwanirathee** [comment]
Prolly I should have skipped pitch shift, speedup and slowdown 😄 . Will try to fix soon, though I am unsure how the libfftw3 issue  will be resolved.

**Comment by ashwanirathee** [comment]
@jparcill can you check where is the issue arising? I haven't changed the code of speedup,slowdown and pitchshift but some issue is there. I asked about the errors in my system of libfftw3 with the core developers of FFTW.jl, and discourse 3-4 days back but nobody has responded yet.

**Comment by jparcill** [comment]
Sure! I'm a bit busy today so I'll check it out tomorrow.

I don't know exactly how github's tests work. Does it test `tests/audio.jl` now?

**Comment by ashwanirathee** [comment]
Things work now, I am not sure why the big duration differences are there. Due to big duration differences I had to specifically set tolerance value high in some test cases.
@jparcill  Yeah it runs tests/audio.jl using the workflow given in .github/workflows

**Comment by jparcill** [comment]
How large are differences?

**Comment by ashwanirathee** [comment]
Below gives an idea for 1 of each case
```julia
    audio_one_channel = SampleBuf(rand(10000), 10)
    audio_two_channel = SampleBuf(rand(10000,2), 10)
    audio_multi_channel = SampleBuf(rand(10000,2), 10)
# In pitchshift case,it's around 2
pitchshift: Test Failed at /home/ashwani/MusicProcessing.jl/test/audio.jl:35
  Expression: isapprox(duration(pitchshift(audio_one_channel)), duration(audio_one_channel), atol = 0)
   Evaluated: isapprox(998.4, 1000.0; atol = 0)

# In speedup case, it's around 40
speedup: Test Failed at /home/ashwani/MusicProcessing.jl/test/audio.jl:43
  Expression: isapprox(duration(speedup(audio_one_channel, 2)), duration(audio_one_channel) / 2.0, atol = 0)
   Evaluated: isapprox(537.6, 500.0; atol = 0)

# In slowdown case, it's big
slowdown: Test Failed at /home/ashwani/MusicProcessing.jl/test/audio.jl:51
  Expression: isapprox(duration(slowdown(audio_one_channel, 2)), duration(audio_one_channel) * 2.0, atol = 0)
   Evaluated: isapprox(1894.4, 2000.0; atol = 0)
```

**Comment by jparcill** [comment]
What happens when you test with an audio file? Maybe that might give us some clues as to whether we're losing data or it's just not speeding up by exact factors

**Comment by ashwanirathee** [comment]
@jparcill No data is lost actually, I was making my samplebuf with very low sample rate which created unrealistic differences. Now when I use 10k sample and sample rate of 1000, differences become fairly small.

File: Project.toml
Comment (Datseris) [inline Project.toml]: If `WAV` is only used for the tests then please do not include it as a dependency of the package. See e.g. https://github.com/JuliaDynamics/DrWatson.jl/blob/master/Project.toml  how the BSON dependency is tests-only.

**Review by jparcill — COMMENTED** [review]
Question about the wav files and commented out tests. Are they for future commits? Because maybe it makes more sense to add them when we need them.

**Review by Datseris — APPROVED** [review]
Once WAV is removed, then feel free to merge (if WAV is used somewhere in the source that I couldn't see, then please ignore me)

**Comment by ashwanirathee** [comment]
I want to keep a single wav file as a general test(sample) file. So when any developer want to test methods locally if things working fine( by auditory method I guess since we don't have definitive tests for most of the cases yet). What do you think?

And the commented out code is setup for using methods on that particular wav file. We might want to keep it or maybe not. 
It's commented out, because it makes the tests fail and it doesn't serve any particular use yet. 

As for WAV.jl, it should be in tests-only package

**Comment by jparcill** [comment]
I think taking out all wav related things out for now makes the most sense.

We can probably add a wav file in another commit. But right now it's not relevant to this current commit.

**Comment by jparcill** [comment]
whoops sorry I'm mobile. will approve in a bit

**Review by jparcill — APPROVED** [review]

**Pull request merged**
